### PR TITLE
os-hypervisor: read offers from openstack plan

### PR DESCRIPTION
**Pull request by gboutry**
Data Source for offers has been added in juju provider 0.7.0, move the offer creation to sunbeam terraform. Read the offers from the plan stored in microcluster.

Depends-On: https://github.com/openstack-snaps/sunbeam-terraform/pull/25

### CODE REVIEW

File: cloud/etc/deploy-openstack-hypervisor/main.tf [hunk 1]
```diff
@@ -26,31 +26,10 @@ terraform {
...
-  endpoint         = "identity-credentials"
-}
-
-resource "juju_offer" "ca_offer" {
-  model            = var.openstack_model
-  application_name = "certificate-authority"
-  endpoint         = "certificates"
-}
+data "terraform_remote_state" "openstack" {
+  backend = "http"
```
Comment (hemanthnakkina) [inline cloud/etc/deploy-openstack-hypervisor/main.tf:30]: All the terraform plans makes no assumptions about the state backend which makes the plans reusable outside snap-openstack if required. The backend configuration only comes from the sunbeam.
Can you make the backend configurable with default to local and sunbeam updates tfvars with backend http.

**Review by hemanthnakkina — CHANGES_REQUESTED** [review]
Thanks for the PR, nice idea to retrieve information from state instead of constructing in the code. Please see comment inline.

File: cloud/etc/deploy-openstack-hypervisor/main.tf (hunk 1 above)
Comment (gboutry) [reply cloud/etc/deploy-openstack-hypervisor/main.tf]: Done

File: cloud/etc/deploy-openstack-hypervisor/main.tf [hunk 2]
```diff
@@ -26,31 +26,9 @@ terraform {
...
-  endpoint         = "certificates"
-}
-
-resource "juju_offer" "ovn_offer" {
-  model            = var.openstack_model
-  application_name = "ovn-relay"
-  endpoint         = "ovsdb-cms-relay"
+data "terraform_remote_state" "openstack" {
+  backend = var.openstack-state-backend
+  config  = var.openstack-state-config
```
Comment (hemanthnakkina) [inline cloud/etc/deploy-openstack-hypervisor/main.tf:31]: As discussed make config generic to work for any backend.

Comment (gboutry) [reply cloud/etc/deploy-openstack-hypervisor/main.tf:31]: made config a freeform map, any config is accepted

**Review by hemanthnakkina — APPROVED** [review]

**Pull request merged**